honeybee_energy/ventcool/crack.py

```diff
@@ -0,0 +1,165 @@
+                * 0.001 - Tight, low-leakage internal wall
+                * 0.0004 - Poor, high-leakage external wall
+                * 0.019 - Poor, high-leakage internal wall
+
+        air_mass_flow_exponent: An optional dimensionless number between 0.5 and 1 used
+            to calculate the crack mass flow rate; required to run an AirflowNetwork
+            simulation. This value represents the leak geometry impact on airflow, with
+            0.5 generally corresponding to turbulent orifice flow and 1 generally
+            corresponding to laminar flow. The default of 0.65 is representative of many
+            cases of wall and window leakage, used when the exponent cannot be measured.
+            (Default: 0.65).
+        crack_factor: A number indicating multiplier for air mass flow through a crack.
+            (Default: 1).
+
+    Properties:
+        * air_mass_flow_coefficient_reference
+        * air_mass_flow_exponent
+        * crack_factor
+        * parent
+        * has_parent
+    """
+    __slots__ = ('_air_mass_flow_coefficient_reference', '_air_mass_flow_exponent',
+                 '_crack_factor', '_parent', '_locked')
+
+    def __init__(self, air_mass_flow_coefficient_reference, air_mass_flow_exponent=0.65,
+                 crack_factor=1):
+        """Initialize AFNCrack."""
+        self.air_mass_flow_coefficient_reference = air_mass_flow_coefficient_reference
+        self.air_mass_flow_exponent = air_mass_flow_exponent
+        self.crack_factor = crack_factor
+        self._parent = None  # this will be set when assigned to an aperture
+
+    @property
+    def air_mass_flow_coefficient_reference(self):
+        """Get or set the air mass flow coefficient defined at the reference crack."""
+        return self._air_mass_flow_coefficient_reference
+
+    @air_mass_flow_coefficient_reference.setter
+    def air_mass_flow_coefficient_reference(self, value):
+        assert value > 0, 'The air_mass_flow_coefficient_reference must be greater ' \
+            'than 0. Got: {}.'.format(value)
+        self._air_mass_flow_coefficient_reference = value
+
+    @property
+    def air_mass_flow_exponent(self):
+        """Get or set the air mass flow exponent for the surface crack."""
+        return self._air_mass_flow_exponent
+
+    @air_mass_flow_exponent.setter
+    def air_mass_flow_exponent(self, value):
+        assert 1 >= value >= 0.5, 'The air_mass_flow_exponent must be greater ' \
+            'than or equal to 0.5 and less than or equal to 1. Got: {}.'.format(value)
+        self._air_mass_flow_exponent = value
+
+    @property
+    def crack_factor(self):
+        """Get or set the multiplier for air mass flow through a crack."""
+        return self._crack_factor
+
+    @crack_factor.setter
+    def crack_factor(self, value):
+        assert 0 < value <= 1, 'The crack_factor must be greater ' \
+            'than 0 and less than or equal to 1. Got: {}.'.format(value)
```

Comment:
I'll just let you fix all of these throughout your code without further comments here.

Reply:
@chriswmackey 

This one actually may pose a problem. In this case the lower bound is greater than 0 but not equal to 0, which is not possible to achieve with the float_in_range function (since it always assumes inclusive ranges (greater/less then or equal to). 

I can fudge this by using an extremely small value for the lower range (0.0000001) or add three new functions in honeybee.typing to account for these kinds of range checks. What do you think?

Reply:
I think the best thing to do is add three new functions to account for different range checks. I'll include it in this module and move it to honeybee.typing if it looks good.

Reply:
I think that's fine if you want to add new functions to honeybee-core, though I think we were finding in your other honeybee-schema PR that we probably want to remove the `crack_factor` and just have people edit the `flow_coefficient`.  So this particular example doesn't require it.

If there are other range checks where you need to exclude the upper/lower value, then you can still add the methods to honeybee-core. I think we should only be strict with this in cases where EnergyPlus fails if a variable is at a certain value, though.